pyanaconda/ui/webui/__init__.py

```diff
@@ -90,25 +88,52 @@ def run(self):
         """Run the interface."""
         log.debug("web-ui: starting cockpit web view")
 
-        # Force Firefox to be used via the BROWSER environment variable.
-        # This is read by cockpit-desktop and makes it launch Firefox in kiosk mode
-        # instead of the GTK WebKit based web view it launches by default.
-
-        # FIXME: looks like "type" should not be used and _is_live_os is private ?
         if conf.system.provides_liveuser:
-            profile_name = FIREFOX_THEME_LIVE
+            self._watch_webui_on_live()
         else:
-            profile_name = FIREFOX_THEME_DEFAULT
+            self._run_webui()
+
+    def _run_webui(self):
+        # FIXME: looks like "type" should not be used and _is_live_os is private ?
+        profile_name = FIREFOX_THEME_DEFAULT
 
         proc = startProgram(["/usr/libexec/webui-desktop",
-                            "-t", profile_name, "-r", str(int(self.remote)),
-                            "/cockpit/@localhost/anaconda-webui/index.html"],
+                             "-t", profile_name, "-r", str(int(self.remote)),
+                             "/cockpit/@localhost/anaconda-webui/index.html"],
                             reset_lang=False)
         log.debug("cockpit web view has been started")
         with open("/run/anaconda/webui_script.pid", "w") as f:
             f.write(repr(proc.pid))
         proc.wait()
-        log.debug("cockpit web view has finished running")
+
+    def _watch_webui_on_live(self):
+        """Watch webui-desktop script process on Live.
+
+        It takes long time to start Firefox after the user interaction (on live even 20+ seconds).
+        To avoid that, we are starting the webui-desktop script early in the liveinst script.
+        Here we are just watching if the process is still running. If the browser is closed
+        Anaconda main process will stop.
+        """
+        log.debug("web-ui: watching webui-desktop pid on live environment")
+        self._main_loop = create_main_loop()
+        pid = -1
+
+        with open("/run/anaconda/webui_script.pid", "tr") as f:
+            pid = int(f.readline().strip())
+
+        PidWatcher.watch_process(pid, self._webui_desktop_closed)
```

Comment:
I think this should be
```
PidWatcher().watch_process(pid, self._webui_desktop_closed)
```

Reply:
Good point. Fixed.

Reply:
This happens if you do changes to the code during creation of tests and not testing it in production afterwards.